lib/currency-utils.js

```diff
@@ -22,23 +22,41 @@ export const getPrecisionFromAmount = amount => {
   return amount.toString().slice(-2) === '00' ? 0 : CurrencyPrecision.DEFAULT;
 };
 
+export const getPrecisionForCurrency = currency => {
+  if (ZeroDecimalCurrencies.includes(currency)) {
+    return 0;
+  }
+
+  return CurrencyPrecision.DEFAULT;
+};
+
+export const getStepForCurrency = currency => {
+  const precision = getPrecisionForCurrency(currency);
+  return 1 / Math.pow(10, precision);
+};
+
 export function formatCurrency(amount, currency = 'USD', options = {}) {
   amount = amount / 100;
-  let minimumFractionDigits = 2;
-  let maximumFractionDigits = 2;
+  let minimumFractionDigits;
+  let maximumFractionDigits;
   if (Object.prototype.hasOwnProperty.call(options, 'minimumFractionDigits')) {
     minimumFractionDigits = options.minimumFractionDigits;
   } else if (Object.prototype.hasOwnProperty.call(options, 'precision')) {
     minimumFractionDigits = options.precision;
     maximumFractionDigits = options.precision;
   }
 
+  if (ZeroDecimalCurrencies.includes(currency) && amount >= 100) {
+    minimumFractionDigits = 0;
+    maximumFractionDigits = 0;
+  }
+
```

Comment:
@Betree This was added to keep the existing behavior of the component, overriding it only when we have a zero decimal currency